howtos/misc/_posts/2021-12-06-178-and-searchbar.md

```diff
@@ -0,0 +1,53 @@
+---
+layout: post
+title:  "PrestaShop 1.7.8 and the search bar issue"
+subtitle: "Search bar problem. How to solve it?"
+date:   2021-12-06 00:00:00
+authors:  [ kpodemski ]
+icon: icon-laptop
+tags: [module, issue, searchbar, upgrade, theme]
+---
+
+We have been informed, that many of you have encountered a problem with the search bar display right after upgrading to PrestaShop 1.7.8.
+This problem also affected installations that were using the standard "classic" theme.
+
+This problem caused some heated discussions in our community, this is why we decided to comment on it and provide you with information about it.
+
+- What is the problem?
+- Why did this happen?
+- How to fix it?
+- What do we plan to do to avoid these types of issues in the future?
+
+## What is the problem?
+
+If you are using a default theme or a theme that overrides the default look of the search bar from the "ps_searchbar" module, after installing the 1.7.8 update you might have seen it displayed like this:
+
+![ps_searchbar issue](/assets/images/2021/12/searchbar_bug.png)
+
+Unfortunately, this problem also affected you when you wanted to skip updates to the default theme.
+
+## Why did this happen?
+
+After one of the last module updates we did, the search bar template file from `/themes/classic/modules/ps_searchbar/ps_searchbar.tpl` was removed from the default theme.
+
+When updating the theme, to avoid major problems, we do not delete its files. We decided to do this because we know that some of the users modify it and it may cause more trouble than benefits. Thus, we came to a situation where one of the files that remained in the default theme was causing the search bar to not display properly.We noticed this problem, and we developed a proper solution for it, but too late.
```

Comment:
```suggestion
To avoid unintended loss of data when performing the upgrade, the update module may overwrite existing files, but will not delete old or "extra" files which are no longer part of the software. Most of the time, leftover files from previous installs will have no negative impact on the upgraded shop. However, in some cases, they do.

In this case, the remaining override in the default theme was causing the search bar to not display properly. This is why the bug only appears when updating from a previous PrestaShop version, and not with a clean shop.
```

Reply:
@atomiix please confirm

Reply:
That's not completly true. The autoupgrade module will delete unused file that was part of the core but not anymore. However, [there is an exception](https://github.com/PrestaShop/autoupgrade/blob/dev/classes/TaskRunner/Upgrade/UpgradeFiles.php#L206-L208): files that are in any `modules/` folder won't be deleted.

Reply:
I updated the suggestion, is it okay now @atomiix ?

Reply:
Okay for me!